### PR TITLE
fix: handle undefined response in apiResponseInterceptor

### PR DESCRIPTION
**Pull request by chrismanley**
An undefined response causes a null ref error in this method rather than raising the expected promise rejection.

### CODE REVIEW

File: src/interceptors/api-response-interceptor.js [hunk 1]
```diff
@@ -11,15 +11,15 @@ import errorCodes from '../error-codes';
...
+  if (response && response.qHandle && response.qType) {
     return session.getObjectApi({
       handle: response.qHandle,
       type: response.qType,
       id: response.qGenericId,
       genericType: response.qGenericType,
     });
   }
-  if (response.qHandle === null && response.qType === null) {
+  if (response && response.qHandle === null && response.qType === null) {
```
Comment (chrismanley) [inline src/interceptors/api-response-interceptor.js:22]: What is the expected shape of the response? If qHandle is undefined is that an error or can we have responses without those? 

I made it so they were errors, but then unit tests failed ... so I'm guessing `null` is relevant here.

Comment (aihazm) [reply src/interceptors/api-response-interceptor.js:22]: the integrations tests were failing due to an expired apiKey in env-vars - now fixed

I see no issue with your changes, but can you explain when that can happen, this is maybe a backend issue? AFAIK, the response should always be there

Comment (T-Wizard) [reply src/interceptors/api-response-interceptor.js:22]: The respone can be the be the value of result or the only out parameter. And therefore almost any type of value.
I think it is possible to have a method where a null response is a valid value but I don't think the engine has any

**Review by T-Wizard — APPROVED** [review]

File: src/interceptors/api-response-interceptor.js (hunk 1 above)
Comment (aihazm) [reply src/interceptors/api-response-interceptor.js:22]: > I think it is possible to have a method where a null response is a valid value but I don't think the engine has any
as @T-Wizard mentioned, afaik engine doesn't have any valid "null" responses

@chrismanley maybe it's worth checking with the engine team

(if none) we should catch & document this error with `createEnigmaError` and add a new entry in the [error-codes.js](https://github.com/qlik-oss/enigma.js/blob/master/src/error-codes.js) ex: `EMPTY_RESPONSE` or similar

**Pull request merged**